src/components/InvertedFlatList/BaseInvertedFlatList.js

```diff
@@ -118,38 +112,40 @@ class BaseInvertedFlatList extends Component {
      *
      * @return {React.Component}
      */
-    renderItem({item, index}) {
-        if (this.props.shouldMeasureItems) {
-            return <View onLayout={({nativeEvent}) => this.measureItemLayout(nativeEvent, index)}>{this.props.renderItem({item, index})}</View>;
-        }
-
-        return this.props.renderItem({item, index});
-    }
-
-    render() {
-        return (
-            <FlatList
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...this.props}
-                ref={this.props.innerRef}
-                renderItem={this.renderItem}
-                // Native platforms do not need to measure items and work fine without this.
-                // Web requires that items be measured or else crazy things happen when scrolling.
-                getItemLayout={this.props.shouldMeasureItems ? this.getItemLayout : undefined}
-                // We keep this property very low so that chat switching remains fast
-                maxToRenderPerBatch={1}
-                windowSize={15}
-
-                // Commenting the line below as it breaks the unread indicator test
-                // we will look at fixing/reusing this after RN v0.72
-                // maintainVisibleContentPosition={{minIndexForVisible: 0, autoscrollToTopThreshold: 0}}
-            />
-        );
-    }
+    const renderItemFromProp = useCallback(
+        ({item, index}) => {
+            if (shouldMeasureItems) {
+                return <View onLayout={({nativeEvent}) => measureItemLayout(nativeEvent, index)}>{renderItem({item, index})}</View>;
+            }
+
+            return renderItem({item, index});
+        },
+        [shouldMeasureItems, measureItemLayout, renderItem],
+    );
+
+    return (
+        <FlatList
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            {...props}
+            ref={innerRef}
+            renderItem={renderItemFromProp}
+            // Native platforms do not need to measure items and work fine without this.
+            // Web requires that items be measured or else crazy things happen when scrolling.
+            getItemLayout={shouldMeasureItems ? getItemLayout : undefined}
+            // We keep this property very low so that chat switching remains fast
+            maxToRenderPerBatch={1}
+            windowSize={15}
+
+            // Commenting the line below as it breaks the unread indicator test
+            // we will look at fixing/reusing this after RN v0.72
+            // maintainVisibleContentPosition={{minIndexForVisible: 0, autoscrollToTopThreshold: 0}}
```

Comment:
I know this comment already existed, but we shouldn't leave TODOs in the code, we should create GH issues for them instead. We have version control, so we also shouldn't leave code commented out
```suggestion

            // Commenting the line below as it breaks the unread indicator test
            // we will look at fixing/reusing this after RN v0.72
            // maintainVisibleContentPosition={{minIndexForVisible: 0, autoscrollToTopThreshold: 0}}
```